SDK/Utils/Reach.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Reach.swift
+//  iOS-SDK
+//
+//  Created by Alberto Huerdo on 4/19/19.
+//  Copyright © 2019 SchedJoules. All rights reserved.
+//
+
+import Foundation
+import SystemConfiguration
+
+
+let ReachabilityStatusChangedNotification = "ReachabilityStatusChangedNotification"
+
+enum ReachabilityType: CustomStringConvertible {
+    case wwan
+    case wiFi
+    
+    var description: String {
+        switch self {
+        case .wwan: return "WWAN"
+        case .wiFi: return "WiFi"
+        }
+    }
+}
+
+enum ReachabilityStatus: CustomStringConvertible  {
+    case offline
+    case online(ReachabilityType)
+    case unknown
+    
+    var description: String {
+        switch self {
+        case .offline: return "Offline"
+        case .online(let type): return "Online (\(type))"
+        case .unknown: return "Unknown"
+        }
+    }
+}
+
+public class Reach {
+    
+    func connectionStatus() -> ReachabilityStatus {
+        var zeroAddress = sockaddr_in()
+        zeroAddress.sin_len = UInt8(MemoryLayout<sockaddr_in>.size)
+        zeroAddress.sin_family = sa_family_t(AF_INET)
+        
+        guard let defaultRouteReachability = withUnsafePointer(to: &zeroAddress, {
+            $0.withMemoryRebound(to: sockaddr.self, capacity: 1) {
+                SCNetworkReachabilityCreateWithAddress(nil, $0)
+            }
+        }) else {
+            return .unknown
+        }
+        
+        var flags : SCNetworkReachabilityFlags = []
+        if !SCNetworkReachabilityGetFlags(defaultRouteReachability, &flags) {
+            return .unknown
+        }
+        
+        return ReachabilityStatus(reachabilityFlags: flags)
+    }
+    
+    
+    func monitorReachabilityChanges() {
+        let host = "google.com"
```

Comment:
why `google.com` and not `api.schedjoules.com`?

Reply:
The idea to look for google is to test the user's connection, and google is almost never down. If we change it to our api and it fails we will give our users the wrong message, that their connection is off while it is our service the one that is down.